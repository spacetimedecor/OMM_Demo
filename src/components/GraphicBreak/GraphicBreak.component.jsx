import {default as React, useEffect, useRef, memo} from "react";
import gsap from "gsap";
import StyledGraphicBreak from "./GraphicBreak.styled";
import PropTypes from "prop-types";
import { useTheme } from 'styled-components'

const GraphicBreak = (props) => {
  const theme = useTheme();
  const {translateX, translateY, duration, ease} = props;

  useEffect(() => {
    gsap
      .to("#graphic-break1", {
        translateX: translateX || theme.graphicBreak.translateX,
        translateY: translateY || theme.graphicBreak.translateY,
        duration: duration || theme.graphicBreak.duration,
        ease: ease || theme.graphicBreak.ease,
        repeat: -1,
        yoyo: true,
      })
  }, [translateX, translateY, duration, ease]);

  return (
    <StyledGraphicBreak
      {...props}
      id="graphic-break1"
    />
  );
};

export default GraphicBreak;

GraphicBreak.propTypes = {
  /**
   * How much the animation animates on the X axis.
   */
  translateX: PropTypes.string,

  /**
   * How much the animation animates on the Y axis.
   */
  translateY: PropTypes.string,

  /**
   * How long the animation takes.
   */
  duration: PropTypes.number,

  /**
   * Animation easing function.
   */
  easing: PropTypes.string,
}
