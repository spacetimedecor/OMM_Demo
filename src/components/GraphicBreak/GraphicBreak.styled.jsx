import styled from "styled-components";
import logo from "../../assets/omm_logo_black.webp";

export const StyledGraphicBreak = styled("div")`
  background-image: url(${logo});
  background-repeat: repeat-x;
  width: ${() => `${96 * 30}px`};
  height: ${() => `${35}px`};
`;

export default StyledGraphicBreak;
