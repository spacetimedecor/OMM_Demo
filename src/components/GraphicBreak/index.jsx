export { default } from "./GraphicBreak.component";
