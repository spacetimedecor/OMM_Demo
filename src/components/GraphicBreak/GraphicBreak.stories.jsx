import GraphicBreak from "./GraphicBreak.component";

import React from "react";

export default {
    title: 'Components/GraphicBreak',
    component: GraphicBreak,
    parameters: {
        layout: 'fullscreen',
    },
    argTypes: {
        translateX: { control: 'text' },
        duration: { control: 'number' }
    }
};

const Template = (args) => <GraphicBreak {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    translateX: "-=10%",
    translateY: "-=0",
    duration: 2,
    ease: "sine.inOut"
};

