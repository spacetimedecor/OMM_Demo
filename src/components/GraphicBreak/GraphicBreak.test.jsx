import * as React from "react";

import themed from "../../theme/themed";
import GraphicBreak from "./GraphicBreak.component";

describe("GraphicBreak", () => {
  it("should render correctly", () => {
    const component = themed(
      <GraphicBreak>
        Test
      </GraphicBreak>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
