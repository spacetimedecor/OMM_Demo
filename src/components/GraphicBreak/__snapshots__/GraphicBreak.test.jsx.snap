// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphicBreak should render correctly 1`] = `
<div
  className="sc-bdvvtL eBsvWT"
  id="graphic-break1"
>
  Test
</div>
`;
